### ISSUE TITLE
Reorg entire repo bc its problematic

### ISSUE THREAD

**Issue by robstryker** [redhat-developer/rsp-server#17]
This repo is poorly structured and is causing problems. Each module is an osgi bundle currently, which also has a main class. Refactoring tools in eclipse often miss some classes, or don't work properly. I have to assume this is due to the projects being maven projects and not recognized completely as tycho projects. The tests are not run in an osgi environment at all, which makes it unclear if they are functioning as they would in an osgi environment. 

In short, a discussion needs to be had about how to actually organize this repository.

**Comment by bdshadow** [comment]
a corresponding JIRA for a tycho build implementation: https://issues.jboss.org/browse/JBIDE-26230

**Comment by robstryker** [comment]
[~dbocharov] did a great job implementing this and did it surprisingly quickly. He took our ideas, added his own, and really stretched himself to get this done. We all appreciate it a lot.
